sbndcode/OpDetSim/sbndPDMapAlg_tool.cc

```
@@ -28,12 +28,23 @@ namespace opdet {
     return false;
   }
 
+  bool sbndPDMapAlg::isElectronics(size_t ch, std::string pdname) const
+  {
+    if(PDmap.at(ch)["electronics"] == std::string(pdname)) return true; // TODO: add number of electronics, daphne01, daphne02, .... ~rodrigoa; at() throws an exception if no such channel
+      return false;
+}
+
   std::string sbndPDMapAlg::pdType(size_t ch) const
   {
-    if(ch < PDmap.size()) return PDmap.at(ch)["pd_type"];
+    if(ch < PDmap.size()) return PDmap.at(ch)["pd_type"];//TODO: remove all if(ch < PDmap.size()) ~rodrigoa
     return "There is no such channel";
```

Can you remove this return please, if it fails to find the channel is better to throw an exception from `at()`